### ISSUE TITLE
aws.client.uploadStorageClass honoring workDir but not publishDir

### ISSUE THREAD

**Issue by rpetit3** [nextflow-io/nextflow#1762]
## Bug report 

I am using AWS Batch with the workDir and publishDir going to S3. I've set `aws.client.uploadStorageClass` to ONEZONE_IA. Most (not all) of the files in the workDir are uploaded to S3 as ONEZONE_IA, but all of the publishDir files are uploaded as STANDARD.

### workDir
![workDir](https://user-images.githubusercontent.com/5334269/96139352-ccfa3900-0ecc-11eb-88ae-9a8b197084d1.JPG)

There are some files that are always uploaded as STANDARD in the workDir (.command.begin, .comman.log, .excitcode) and not the expected storage class.

### publishDir
![publishDir](https://user-images.githubusercontent.com/5334269/96139443-e26f6300-0ecc-11eb-9fd8-8402148ed8c3.JPG)
![publishDir2](https://user-images.githubusercontent.com/5334269/96139452-e56a5380-0ecc-11eb-8c97-0ef425750f0e.JPG)

In the images above there is an example of a file (DRX100185.ctx) being ONEZONE_IA in the workDir then STANDARD in publishDir. Then the other image is just to show another set of outputs also being STANDARD only.

My publishDir Directives follow this pattern:  
```
publishDir "${outdir}/${sample}", mode: "${params.publish_mode}", overwrite: params.overwrite, pattern: '*.txt'
```

With `outdir` set to `s3://BUCKET_NAME/`, `params.publish_mode` set to `copy`, and `params.overwrite` to `true`.

I am running Bactopia, available here: https://github.com/bactopia/bactopia/

### Expected behavior and actual behavior

All files uploaded to S3 should use the storage class specified in the config file.

### Steps to reproduce the problem

In my case at least, set workDir to 's3://BUCK_NAME/work/test-100' and publishDir 's3://BUCKET_NAME/results'

(I've also tried making them two different buckets, but still had the same issue)

### Program output 

Doesn't really fail, but I can see ONEZONE_IA is set in the log
```
grep ONEZONE .nextflow.log
Oct-14 15:35:03.266 [main] DEBUG nextflow.file.FileHelper - AWS S3 config details: {secret_key=..., upload_storage_class=ONEZONE_IA, region=us-east-1, access_key=...}
Oct-14 15:35:19.975 [main] DEBUG n.cloud.aws.batch.AwsBatchExecutor - [AWS BATCH] Executor options=AwsOptions(cliPath:/home/ec2-user/conda/bin/aws, storageClass:ONEZONE_IA, storageEncryption:null, remoteBinDir:/staphopia-v2/work/test-1/tmp/c8/28737ef72683c2e21a29ccc0aec56b/bin, region:us-east-1, maxParallelTransfers:8, maxTransferAttempts:3, delayBetweenAttempts:15ms, fetchInstanceType:true, jobRole:null, volumes:[/opt/conda:/mnt/conda], awsCli:/home/ec2-user/conda/bin/aws --region us-east-1)
```

### Environment 

* Nextflow version: 20.07.1 build 5412
* Java version: openjdk 11.0.8-internal 2020-07-14
* Operating system: Linux (Debian and Amazon Linux2)
* Bash version: 4.2.46(2)-release

_Note: Nextflow and Java are conda builds._

### Additional context

I haven't tried other storage classes. I have tower running if you like to take a look at the run details. Also happy to share anything/information you think would be useful.

Thank you!

**Comment by jmarti-n1h** [comment]
BUG FOLLOW UP
--
As of 2023, Nextflow version 23.04.3 build 5875, the issue persists (or it is even worse), albeit now changing `aws.client.uploadStorageClass = 'ONEZONE_IA' ` to 

```
aws{
                region = 'eu-west-3'
                batch.cliPath = '/home/ec2-user/miniconda/bin/aws'
                executor.awscli = '/home/ec2-user/miniconda/bin/aws'
                client.uploadStorageClass = 'INTELLIGENT_TIERING'
}
```

Contents of `.nextflow.log` : `Aug-24 11:54:09.034 [main] DEBUG n.cloud.aws.batch.AwsBatchExecutor - [AWS BATCH] Executor options=AwsOptions(cliPath:/home/ec2-user/miniconda/bin/aws, storageClass:null, storageEncryption:null, storageKmsKeyId:null, remoteBinDir:null, region:null, maxParallelTransfers:4, maxTransferAttempts:5, delayBetweenAttempts:10s, retryMode:standard, maxSpotAttempts:5, debug:null, fetchInstanceType:false, jobRole:null, logsGroup:null, volumes:[], shareIdentifier:null, schedulingPriority:0, s3Acl:null, awsCli:/home/ec2-user/miniconda/bin/aws)`

WorkDir contents:
![image](https://github.com/nextflow-io/nextflow/assets/101389426/aba10eef-3905-4a3a-ae7c-83a665fefa9d)

PublishDir contents:
![image](https://github.com/nextflow-io/nextflow/assets/101389426/93b245e5-2856-41af-a456-138d831ea737)

Seems that for storageClass as INTELLIGENT_TIERING is not being recognized, sent or working.

Thanks!

**Comment by bentsherman** [comment]
This issue was fixed by https://github.com/nextflow-io/nextflow/commit/066f9203d8f60ff86677697c2e8c941091f1dced which is available since 22.12.0-edge. You can now set the `storageClass` option for `publishDir`.
